### PR TITLE
comments api

### PR DESCRIPTION
**Pull request by Dhruv8321**

### CODE REVIEW

File: src/middleware/commentMiddleware.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const express = require('express');
+const app = express();
+
+app.use(function (req, res) {
+
+      console.log('comments posted');
+      next();
+ })
+
```
Comment (vibhor-kukreja) [inline src/middleware/commentMiddleware.js:9]: Not required

File: src/models/commentSchema.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      ref: 'User'}, 
+    feed_Id: {
+      type: mongoose.Schema.Types.ObjectId, 
+      ref: 'Feed'},
+    updated_At: {
+      type: Date, default: new Date()},
+    created_At: {
+      type: Date, default: new Date()} 
+
+
```
Comment (vibhor-kukreja) [inline src/models/commentSchema.js:20]: remove

File: src/models/commentSchema.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const mongoose = require('mongoose');
+const Comment = mongoose.model('Comment', new mongoose.Schema({
+
+    comment: {
+      type: String, 
+      minlength: 2,
```
Comment (vibhor-kukreja) [inline src/models/commentSchema.js:6]: Min should be 5 to 10 char

File: src/models/commentSchema.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+const mongoose = require('mongoose');
+const Comment = mongoose.model('Comment', new mongoose.Schema({
```
Comment (vibhor-kukreja) [inline src/models/commentSchema.js:2]: Should be comments

**Review by vibhor-kukreja — CHANGES_REQUESTED** [review]

File: src/middleware/Validation.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+const Joi = require('joi');
+
+//Joi Validation
+function validateComment(comment) {
+  const schema = {
+    comment: Joi.string().min(5).max(10).required()
```
Comment (ayushsaxena2022) [inline src/middleware/Validation.js:6]: comment can be more than 10 characters

File: src/models/friendlist.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  friend: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true
+  },
+
+  username: {
+
+    type: String,
+
```
Comment (ayushsaxena2022) [inline src/models/friendlist.js:17]: remove extra lines

File: src/models/comment.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  comment: {
+    type: String,
+    minlength: 5,
+    maxlength: 10,
+  },
+  user_Id: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'User'
+  },
+  feed_Id: {
```
Comment (ayushsaxena2022) [inline src/models/comment.js:13]: use one type of casing

File: src/routes/comments.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+const express = require('express');
+const router = require('express').Router();
+const comment = require('../models/comment');
+const myfeed = require('../models/myfeed');
+
+
+
+//Create Comment
+
+router.post('/comments/:feed_Id', async (req, res) => {
```
Comment (ayushsaxena2022) [inline src/routes/comments.js:11]: extra lines remove

File: src/routes/comments.js [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+   try {
+        const feed_Id = await myfeed.findOne({ _id: req.params.feed_Id });
+        const comments = new comment();
+        comments.comment = req.body.comment;
+        comments.feed_Id = feed_Id._id;
+        await comments.save();
+        feed_Id.User.push(comments._id);
+        await post.save();
```
Comment (ayushsaxena2022) [inline src/routes/comments.js:19]: no need of post.save

Comment (gurjot74) [inline src/routes/comments.js:19]: post is not defined

**Pull request merged**